### PR TITLE
KAS-4606: Agenda activity numbering migration

### PR DESCRIPTION
**Pull request by sergiofenoll**
https://kanselarij.atlassian.net/browse/KAS-4606

Migration service to add the "agenderingsactiviteitnummer" to subcases, as well backfill the order and "original name" of pieces on agenda's of 2024.

This service lists the subcases in order of "agendering" starting in January of 2024 and then checks their containing pieces to 1) ensure the numbering as performed by the secretary is correct and 2) apply said numbering in a structured way using triples.

To test this, it's necessary to take a copy of the production database locally and perform the migration on it. The migration will fail on our own test data (because our documents do not follow the secretary's naming convention) as well as on data from the ACC database, as the numbering isn't respected in the testing agendas.

Verification of the applied numbering can be done in the triplestore, by querying the changed subcases/pieces and checking, or by using the following snippet in the frontend code to see the numbers in the app:

```diff
diff --git a/app/components/agenda/agendaitem/agendaitem-case-panel/agendaitem-case-panel-view.hbs b/app/components/agenda/agendaitem/agendaitem-case-panel/agendaitem-case-panel-view.hbs
index 715f6359d..bb6305ff4 100644
--- a/app/components/agenda/agendaitem/agendaitem-case-panel/agendaitem-case-panel-view.hbs
+++ b/app/components/agenda/agendaitem/agendaitem-case-panel/agendaitem-case-panel-view.hbs
@@ -49,7 +49,7 @@
         class="auk-o-flex auk-o-flex--justify-between auk-o-flex--vertical-center"
       >
         <h4 data-test-agendaitem-titles-short-title class="auk-u-m-0">
-          {{@agendaitem.shortTitle}}
+          {{@agendaitem.shortTitle}} - agenda activity number: {{@subcase.agendaActivityNumber}}
         </h4>
         <div
           class="auk-o-flex auk-o-flex--vertical-center auk-o-flex--spaced"
diff --git a/app/components/documents/document-card.hbs b/app/components/documents/document-card.hbs
index a9e7a75b1..695014bf3 100644
--- a/app/components/documents/document-card.hbs
+++ b/app/components/documents/document-card.hbs
@@ -53,7 +53,7 @@
                   )
                   as |extension|
                 }}
-                  {{this.piece.name}}{{#if extension}}.{{extension}}{{/if}}
+                  {{this.piece.name}}{{#if extension}}.{{extension}}{{/if}} - position: {{this.piece.position}}
                 {{/let}}
               {{/if}}
             </LinkTo>
diff --git a/app/components/subcase/description-panel/view.hbs b/app/components/subcase/description-panel/view.hbs
index dd9beb1d6..063fc35fb 100644
--- a/app/components/subcase/description-panel/view.hbs
+++ b/app/components/subcase/description-panel/view.hbs
@@ -32,7 +32,7 @@
           @level="4"
           @skin="4"
         >
-          {{capitalize @subcase.shortTitle}}
+          {{capitalize @subcase.shortTitle}} - agenda activity number: {{@subcase.agendaActivityNumber}}
         </AuHeading>
         <div class="au-u-flex au-u-flex--spaced-tiny au-u-flex--vertical-center">
           <AuPill

```

### CODE REVIEW

File: app.js [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+      pieces.push({ piece, pieceName, vrDoc });
+
+      // Validation
+      if (isDoc && isMed) {
+        addError({ subcase, counter, vrDoc, message: 'Marked as a NOTA and a MEDEDLING' });
+      }
+
+      if (isDec && isMed) {
+        addError({ subcase, counter, vrDoc, message: 'Marked as a DECREET and a MEDEDELING' });
+      }
```
Comment (kobemertens) [inline app.js]: Can't this be taken out of this loop? Also don't see how `isDoc` and `isMed` can be true at the same time

File: app.js [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+      } else {
+        isDec
+          ? counters.regular.dec++
+          : isDoc
+            ? counters.regular.doc++
+            : counters.regular.med++;
+      }
+    }
+
+    for (const [index, { vrDoc }] of pieces.sort((a, b) => a.vrDoc.meta.index - b.vrDoc.meta.index).entries()) {
```
Comment (kobemertens) [inline app.js]: Might be more readable to put the iterable in a separate variable

File: app.js [hunk 3]
```diff
@@ -0,0 +1,141 @@
...
+      }
+    }
+
+    if (errors.length) {
+      break;
+    }
+
+    // Add new data
+    await addAgendaSubcaseId(subcase, counter);
+    for (const [index, { piece, pieceName }] of pieces.sort((a, b) => a.vrDoc.meta.index - b.vrDoc.meta.index).entries()) {
```
Comment (kobemertens) [inline app.js]: Could use the stored variable here

**Review by kobemertens — CHANGES_REQUESTED** [review]
Looks good, just some minor things that could be cleaned up

**Review by tdn — APPROVED** [review]
Made some minor changes, such as adding position to the document container instead of the piece. Also added an env variable to ensure this can be run locally, where the filenames are not matched to the real world situation.

**Pull request merged**